### ISSUE TITLE
Add Aruco-marker detection using opencv

### ISSUE THREAD

**Issue by vishapraj** [abhisheks008/ML-Crate#551]
### ML-Crate Repository (Proposing new issue)
:red_circle: **Project Title** :
:red_circle: **Aim** :
:red_circle: **Dataset** :
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Participant ID (If not, then put NA) :
- Approach for this Project :
- What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by abhisheks008** [comment]
Need to share the dataset and approach for your project.
@vishapraj

**Comment by AbhineshJha** [comment]
Dataset: 
![image](https://github.com/abhisheks008/ML-Crate/assets/142514166/ddad4ec5-dbdf-49c3-a708-3da90b328f93)

Approach : 

Preprocessing:

Resize, crop, and adjust brightness/contrast.
Detection:

Use OpenCV's cv2.aruco.detectMarkers for marker identification.
Pose Estimation (Optional):

Implement 3D pose estimation with cv2.aruco.estimatePoseSingleMarkers.
Visualization & Evaluation:

Visualize and evaluate marker detection accuracy.
Optimization:

Fine-tune parameters for better performance.



can u assign me with this issue under IWOC .! @abhisheks008

**Comment by abhisheks008** [comment]
I have updated the issue name, follow the same issue name for project folder as well as pull request.

Issue assigned to you @vishapraj under IWOC 2024.

**Comment by AbhineshJha** [comment]
Assign me  @abhisheks008 🙃

**Comment by abhisheks008** [comment]
> Assign me @abhisheks008 🙃

Oh shoot! Sorry.

Assigned to @AbhineshJha under IWOC

**Comment by abhisheks008** [comment]
Unassigned as the open source event ended up.

**Comment by jayeshrdeotalu** [comment]
can you assign this to me @abhisheks008 

Approach will be same :

Preprocessing:

Resize, crop, and adjust brightness/contrast.
Detection:

Use OpenCV's cv2.aruco.detectMarkers for marker identification.

Visualization & Evaluation:

Visualize and evaluate marker detection accuracy.

**Comment by abhisheks008** [comment]
> can you assign this to me @abhisheks008
> 
> Approach will be same :
> 
> Preprocessing:
> 
> Resize, crop, and adjust brightness/contrast. Detection:
> 
> Use OpenCV's cv2.aruco.detectMarkers for marker identification.
> 
> Visualization & Evaluation:
> 
> Visualize and evaluate marker detection accuracy.

In which open source program/event are you participating in?

**Comment by jayeshrdeotalu** [comment]
None... just want to contribute

**Comment by abhisheks008** [comment]
As this project repository is currently part of different open source events, you can contribute here as an individual contributor after Feb 29th, 2024.
@jayeshrdeotalu

**Comment by jayeshrdeotalu** [comment]
Okay...

**Comment by jayeshrdeotalu** [comment]
So, can you assign it to me now...?

**Comment by abhisheks008** [comment]
Assigned to you as a contributor @jayeshrdeotalu

**Comment by milanprajapati571** [comment]
Full name: Milan Prajapati
GitHub Profile Link: https://github.com/milanprajapati571
Participant ID (If not, then put NA): NA
Approach for this Project: Develop an ArUco marker detection project using OpenCV for image processing, Python for scripting, and TensorFlow for potential machine learning enhancements.
What is your participant role? : SSoC (Social Summer of Code)

Sir, can You Please assign this project to me...?

**Comment by abhisheks008** [comment]
What are the models you are planning for this dataset? Brief your approach with 3-4 models.

**Comment by aryamanpathak2022** [comment]
**Full name**: Aryaman Pathak 

**GitHub Profile Link**: [Profile](https://github.com/aryamanpathak2022)

**Participant ID**: NA

**Approach for this Project**:
My approach will include:
1. **Exploratory Data Analysis (EDA)**: Understanding the dataset through visualizations and summary statistics.
2. **Data Preprocessing**: Cleaning and preparing the data for analysis, including tokenization, removing stopwords, and other text preprocessing techniques.
3. **Model Implementation**: Implementing 3-4 machine learning algorithms such as Logistic Regression, Random Forest, SVM, and Naive Bayes for sentiment analysis.
4. **Model Comparison**: Comparing the performance of the models using accuracy scores and other relevant metrics to determine the best-fit model.
5. **Documentation**: Documenting the entire process, including the EDA, preprocessing steps, model implementations, comparisons, and conclusions in the README.md file.

**What is your participant role?**: SSOC (Social Summer of Code)

**Comment by abhisheks008** [comment]
Implement these models for this project,
1. Random Forest
2. Decision Tree
3. Logistic Regression
4. Gradient Boosting
5. XGBoost
6. Lasso
7. Ridge
8. MLP Classifier

Assigned @aryamanpathak2022

**Comment by aryamanpathak2022** [comment]
Hi @abhisheks008 

I have unassigned myself from this project due to other commitments that require my immediate attention. I apologize for any inconvenience this may cause.
